iati/tests/test_resources.py

```diff
@@ -50,193 +177,492 @@ class TestResourceFolders(object):
+        """Check that a ValueError is raised when trying to create a path for an unknown version of the IATI Standard."""
+        with pytest.raises(ValueError):
+            iati.resources.path_for_version(filename_no_meaning_single, std_ver_all_mixedinst_valid_unknown)
 
-        assert iati.resources.BASE_PATH_LIB_DATA != ''
-        assert iati.resources.BASE_PATH_LIB_DATA in path
-        assert file_name == path[-len(file_name):]
+    def test_path_for_version_unknown_ver_typeerr(self, filename_no_meaning_single, std_ver_all_uninst_typeerr):
+        """Check that a TypeError is raised when trying to create a folder path for a value of a type that cannot be a version number."""
+        with pytest.raises(TypeError):
+            iati.resources.path_for_version(filename_no_meaning_single, std_ver_all_uninst_typeerr)
 
+    def test_path_for_version_path_valueerr(self, filepath_invalid_value, std_ver_minor_inst_valid_single):
+        """Check that a ValueError is raised when trying to create a path from a string that cannot be a filepath."""
+        with pytest.raises(ValueError):
+            iati.resources.path_for_version(filepath_invalid_value, std_ver_minor_inst_valid_single)
 
-class TestResourceCodelists(object):
-    """A container for tests relating to Codelist resources."""
+    def test_path_for_version_path_typeerr(self, filepath_invalid_type, std_ver_minor_inst_valid_single):
+        """Check that a TypeError is raised when trying to create an absolute path from a path of an incorrect type."""
+        with pytest.raises(TypeError):
+            iati.resources.path_for_version(filepath_invalid_type, std_ver_minor_inst_valid_single)
 
-    def test_codelist_flow_type(self, std_ver_minor_mixedinst_valid_fullsupport):
-        """Check that the FlowType codelist is loaded as a string and contains content."""
-        path = iati.resources.create_codelist_path('FlowType', std_ver_minor_mixedinst_valid_fullsupport)
 
-        content = iati.utilities.load_as_string(path)
+@pytest.mark.new_tests
+class TestResourcePathCreationCodelistMapping(object):
+    """A container for tests relating to creating Codelist Mapping File paths."""
 
-        assert len(content) > 3200
-        assert iati.validator.is_xml(content)
+    def test_create_codelist_mapping_path_minor(self, std_ver_minor_mixedinst_valid_known):
+        """Check that there is a single Codelist Mapping File for minor versions."""
+        version_folder = iati.resources.folder_name_for_version(std_ver_minor_mixedinst_valid_known)
+        path = iati.resources.create_codelist_mapping_path(std_ver_minor_mixedinst_valid_known)
 
-    def test_find_codelist_paths(self, codelist_lengths_by_version):
-        """Check that all codelist paths are being found."""
-        paths = iati.resources.get_codelist_paths(codelist_lengths_by_version[0])
+        assert isinstance(path, str)
```

Comment:
Should do more checks in general to ensure that a resulting value is a string (ideally `PathLike`, but 2.7 :( ) - returning a list rather than a string may not be caught in a number of other places.